### ISSUE TITLE
Segmentation fault opening database

### ISSUE THREAD

**Issue by Generator** [keepassxreboot/keepassxc#1156]
## Expected Behavior
Open my database.

## Current Behavior
KeePassXC ends in segmentation fault when try to open database.

## Steps to Reproduce (for bugs)

Not very clean, only happens with my database, issue is with some entry.

## Context

With my database KeePassXC always ends in segmentation fault, creating a new database works, i've created a new database with same credentials of my database and then merge the old database with new on, KeePassXC continues to works without issue, but after close and reopen the new database same issue, crash on open database.

## Debug Info

KeePassXC - Versão 2.2.2
Revisão: e100ba8

Bibliotecas:
- Qt 5.9.2
- libgcrypt 1.8.1

Sistema operativo: Arch Linux
Arquitetura do CPU: x86_64
Kernel: linux 4.13.10-1-ck-core2

Extensões ativas:
- KeePassHTTP
- Auto-Type
- YubiKey

## Notes
No other patches applied, compiled only with develop branch.

Tried to do some debug with gdb:
```
[New Thread 0x7f8fa9138700 (LWP 28576)]
[New Thread 0x7f8fa8937700 (LWP 28577)]
[New Thread 0x7f8f9fde2700 (LWP 28762)]
[New Thread 0x7f8f9f5e1700 (LWP 28763)]
[New Thread 0x7f8f9ede0700 (LWP 28764)]
[New Thread 0x7f8f9d7b9700 (LWP 28765)]
[New Thread 0x7f8f9cfb8700 (LWP 28766)]
[New Thread 0x7f8f87fff700 (LWP 28767)]
[Thread 0x7f8f9d7b9700 (LWP 28765) exited]
[Thread 0x7f8f9cfb8700 (LWP 28766) exited]
[Thread 0x7f8f9f5e1700 (LWP 28763) exited]
[Thread 0x7f8f87fff700 (LWP 28767) exited]
[Thread 0x7f8f9fde2700 (LWP 28762) exited]

Thread 1 "keepassxc" received signal SIGSEGV, Segmentation fault.
0x00007f8fd1d02db0 in QTimer::stop() () from /usr/lib/libQt5Core.so.5
```

**Comment by TheZ3ro** [comment]
We need more information.

Try making a copy of that database and delete entry one by one, checking which entry makes the DB crash.
Once you found that entry try copy its value (password, username, url) in a new database one by one so you can detect which field make the db crash.

**Comment by Generator** [comment]
Looks like the issue was with my old `KeePassHttp Settings`, originally created by KeePassHttp addon to KeePass. 

Deleting the entry (and empty trash **<- required or will crash again**) and then create a new with browser addon (e.g. KeePassHttp-Connector) solved the issue, i able to open my database again without crashing.

**Comment by adolfogc** [comment]
## Bug report

### Steps to reproduce
1. No problem:
- Compile using the `develop` branch
- Run KeePassXC
- Create a new DB
- Create an entry
- Save and close
- Try to open the DB [no segfault]

2. Modify the DB so it segfaults later:
- Set-up TOTP using: `otpauth://totp/Test?secret=kxvnvtdzcrb6zbheofs3a5vr4c7soxkv&algorithm=SHA1&digits=6&period=30` as a seed
- Save and close
- Try to open the DB [segfaults]

### Build command
```
mkdir build && cd build
cmake .. -DCMAKE_BUILD_TYPE=Debug -DWITH_ASAN=ON
make -j4
```

### Debug info.
KeePassXC - Version 2.2.2
Revision: 2106be6

Libraries:
- Qt 5.9.2
- libgcrypt 1.8.1

Operating system: macOS High Sierra (10.13)
CPU architecture: x86_64
Kernel: darwin 17.0.0

Enabled extensions:
- Auto-Type

### Output from ASan on macOS
```
ASAN:DEADLYSIGNAL
=================================================================
==32128==ERROR: AddressSanitizer: SEGV on unknown address 0x000000000000 (pc 0x00010ff81859 bp 0x7ffee11d7eb0 sp 0x7ffee11d7ea0 T0)
==32128==The signal is caused by a READ memory access.
==32128==Hint: address points to the zero page.
    #0 0x10ff81858 in QTimer::stop() (QtCore:x86_64+0x221858)
    #1 0x10eb9a1ce  (KeePassXC:x86_64+0x10017b1ce)
    #2 0x10eb9855d  (KeePassXC:x86_64+0x10017955d)
    #3 0x10ed15511  (KeePassXC:x86_64+0x1002f6511)
    #4 0x10ff7a13a in QMetaObject::activate(QObject*, int, int, void**) (QtCore:x86_64+0x21a13a)
    #5 0x10ed145d3  (KeePassXC:x86_64+0x1002f55d3)
    #6 0x10eb81224  (KeePassXC:x86_64+0x100162224)
    #7 0x10eb811c2  (KeePassXC:x86_64+0x1001621c2)
    #8 0x10ed10045  (KeePassXC:x86_64+0x1002f1045)
    #9 0x10ff7a13a in QMetaObject::activate(QObject*, int, int, void**) (QtCore:x86_64+0x21a13a)
    #10 0x10ed24d94  (KeePassXC:x86_64+0x100305d94)
    #11 0x10ed2429f  (KeePassXC:x86_64+0x10030529f)
    #12 0x10ff7a13a in QMetaObject::activate(QObject*, int, int, void**) (QtCore:x86_64+0x21a13a)
    #13 0x10ff09d9a in QItemSelectionModel::emitSelectionChanged(QItemSelection const&, QItemSelection const&) (QtCore:x86_64+0x1a9d9a)
    #14 0x10ff09a6a in QItemSelectionModel::select(QItemSelection const&, QFlags<QItemSelectionModel::SelectionFlag>) (QtCore:x86_64+0x1a9a6a)
    #15 0x10ff09767 in QItemSelectionModel::select(QModelIndex const&, QFlags<QItemSelectionModel::SelectionFlag>) (QtCore:x86_64+0x1a9767)
    #16 0x10ff0a539 in QItemSelectionModel::setCurrentIndex(QModelIndex const&, QFlags<QItemSelectionModel::SelectionFlag>) (QtCore:x86_64+0x1aa539)
    #17 0x10ecc216b  (KeePassXC:x86_64+0x1002a316b)
    #18 0x10ecc1c81  (KeePassXC:x86_64+0x1002a2c81)
    #19 0x10ecc17b1  (KeePassXC:x86_64+0x1002a27b1)
    #20 0x10eb80a9c  (KeePassXC:x86_64+0x100161a9c)
    #21 0x10ed0eb2c  (KeePassXC:x86_64+0x1002efb2c)
    #22 0x10ff7a13a in QMetaObject::activate(QObject*, int, int, void**) (QtCore:x86_64+0x21a13a)
    #23 0x10ed26fc0  (KeePassXC:x86_64+0x100307fc0)
    #24 0x10ecd7234  (KeePassXC:x86_64+0x1002b8234)
    #25 0x10ed268d1  (KeePassXC:x86_64+0x1003078d1)
    #26 0x10ff7a13a in QMetaObject::activate(QObject*, int, int, void**) (QtCore:x86_64+0x21a13a)
    #27 0x10ff0a575 in QItemSelectionModel::setCurrentIndex(QModelIndex const&, QFlags<QItemSelectionModel::SelectionFlag>) (QtCore:x86_64+0x1aa575)
    #28 0x10f44ce34 in QAbstractItemView::setCurrentIndex(QModelIndex const&) (QtWidgets:x86_64+0x26fe34)
    #29 0x10ecd5ffb  (KeePassXC:x86_64+0x1002b6ffb)
    #30 0x10ed26b02  (KeePassXC:x86_64+0x100307b02)
    #31 0x10ff7a13a in QMetaObject::activate(QObject*, int, int, void**) (QtCore:x86_64+0x21a13a)
    #32 0x10fefc3c8 in QAbstractItemModel::endResetModel() (QtCore:x86_64+0x19c3c8)
    #33 0x10eccdf0c  (KeePassXC:x86_64+0x1002aef0c)
    #34 0x10ecd6141  (KeePassXC:x86_64+0x1002b7141)
    #35 0x10eb76a8b  (KeePassXC:x86_64+0x100157a8b)
    #36 0x10eb7da58  (KeePassXC:x86_64+0x10015ea58)
    #37 0x10ed10313  (KeePassXC:x86_64+0x1002f1313)
    #38 0x10ff7a13a in QMetaObject::activate(QObject*, int, int, void**) (QtCore:x86_64+0x21a13a)
    #39 0x10ed07b61  (KeePassXC:x86_64+0x1002e8b61)
    #40 0x10eb4a2f1  (KeePassXC:x86_64+0x10012b2f1)
    #41 0x10ed07596  (KeePassXC:x86_64+0x1002e8596)
    #42 0x10ff7a13a in QMetaObject::activate(QObject*, int, int, void**) (QtCore:x86_64+0x21a13a)
    #43 0x10f389aa7  (QtWidgets:x86_64+0x1acaa7)
    #44 0x10ff7a13a in QMetaObject::activate(QObject*, int, int, void**) (QtCore:x86_64+0x21a13a)
    #45 0x10f2e6a6e  (QtWidgets:x86_64+0x109a6e)
    #46 0x10f2e6906  (QtWidgets:x86_64+0x109906)
    #47 0x10f2e7b6f in QAbstractButton::mouseReleaseEvent(QMouseEvent*) (QtWidgets:x86_64+0x10ab6f)
    #48 0x10f2288a9 in QWidget::event(QEvent*) (QtWidgets:x86_64+0x4b8a9)
    #49 0x10f1eedfc in QApplicationPrivate::notify_helper(QObject*, QEvent*) (QtWidgets:x86_64+0x11dfc)
    #50 0x10f1f1b8c in QApplication::notify(QObject*, QEvent*) (QtWidgets:x86_64+0x14b8c)
    #51 0x10ff49bd7 in QCoreApplication::notifyInternal2(QObject*, QEvent*) (QtCore:x86_64+0x1e9bd7)
    #52 0x10f1ef75f in QApplicationPrivate::sendMouseEvent(QWidget*, QMouseEvent*, QWidget*, QWidget*, QWidget**, QPointer<QWidget>&, bool) (QtWidgets:x86_64+0x1275f)
    #53 0x10f2496bf  (QtWidgets:x86_64+0x6c6bf)
    #54 0x10f248354  (QtWidgets:x86_64+0x6b354)
    #55 0x10f1eedfc in QApplicationPrivate::notify_helper(QObject*, QEvent*) (QtWidgets:x86_64+0x11dfc)
    #56 0x10f1f0199 in QApplication::notify(QObject*, QEvent*) (QtWidgets:x86_64+0x13199)
    #57 0x10ff49bd7 in QCoreApplication::notifyInternal2(QObject*, QEvent*) (QtCore:x86_64+0x1e9bd7)
    #58 0x10f88c28a in QGuiApplicationPrivate::processMouseEvent(QWindowSystemInterfacePrivate::MouseEvent*) (QtGui:x86_64+0x2e28a)
    #59 0x10f87394a in QWindowSystemInterface::sendWindowSystemEvents(QFlags<QEventLoop::ProcessEventsFlag>) (QtGui:x86_64+0x1594a)
    #60 0x1163a702d  (libqcocoa.dylib:x86_64+0x2f02d)
    #61 0x1163a78f0  (libqcocoa.dylib:x86_64+0x2f8f0)
    #62 0x7fff2d64b6c0 in __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ (CoreFoundation:x86_64h+0xa36c0)
    #63 0x7fff2d7052cb in __CFRunLoopDoSource0 (CoreFoundation:x86_64h+0x15d2cb)
    #64 0x7fff2d62e15f in __CFRunLoopDoSources0 (CoreFoundation:x86_64h+0x8615f)
    #65 0x7fff2d62d5dc in __CFRunLoopRun (CoreFoundation:x86_64h+0x855dc)
    #66 0x7fff2d62ce42 in CFRunLoopRunSpecific (CoreFoundation:x86_64h+0x84e42)
    #67 0x7fff2c94c865 in RunCurrentEventLoopInMode (HIToolbox:x86_64+0x2f865)
    #68 0x7fff2c94c4de in ReceiveNextEventCommon (HIToolbox:x86_64+0x2f4de)
    #69 0x7fff2c94c353 in _BlockUntilNextEventMatchingListInModeWithFilter (HIToolbox:x86_64+0x2f353)
    #70 0x7fff2ac499f6 in _DPSNextEvent (AppKit:x86_64+0x419f6)
    #71 0x7fff2b3ded97 in -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] (AppKit:x86_64+0x7d6d97)
    #72 0x7fff2ac3e804 in -[NSApplication run] (AppKit:x86_64+0x36804)
    #73 0x1163a66dc  (libqcocoa.dylib:x86_64+0x2e6dc)
    #74 0x10ff4598d in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) (QtCore:x86_64+0x1e598d)
    #75 0x10ff4a2a0 in QCoreApplication::exec() (QtCore:x86_64+0x1ea2a0)
    #76 0x10ea25798  (KeePassXC:x86_64+0x100006798)
    #77 0x7fff54b94144 in start (libdyld.dylib:x86_64+0x1144)

==32128==Register values:
rax = 0x000060c000354818  rbx = 0xbebebebebebebebe  rcx = 0x000060c000354818  rdx = 0x00001c180006a903  
rdi = 0xbebebebebebebebe  rsi = 0x000060c0003547c0  rbp = 0x00007ffee11d7eb0  rsp = 0x00007ffee11d7ea0  
 r8 = 0x00001fffdc23afd8   r9 = 0x00001c180006a8ff  r10 = 0x00007ffee11d7ee0  r11 = 0x00007ffee11d7f00  
r12 = 0x00007ffee11d7f60  r13 = 0xf2f2f2f8f2f2f200  r14 = 0x00007ffee11d7f20  r15 = 0x00007ffee11d7f40  
AddressSanitizer can not provide additional info.
SUMMARY: AddressSanitizer: SEGV (QtCore:x86_64+0x221858) in QTimer::stop()
==32128==ABORTING
Abort trap: 6
```

## Another observation
I can use the latest released version (2.2.2) to open the "buggy" DB without any problem.

### Debug info.
KeePassXC - Version 2.2.2
Revision: 6d46717

Libraries:
- Qt 5.9.1
- libgcrypt 1.8.1

Operating system: macOS High Sierra (10.13)
CPU architecture: x86_64
Kernel: darwin 17.0.0

Enabled extensions:
- KeePassHTTP
- Auto-Type
- YubiKey

## One final comment
❗️ Please notice that the OP uses e100ba8 but that I'm using 2106be6.

**Comment by TheZ3ro** [comment]
This is due to the DetailsWidget that parse the TOTP and apparently "fail".
I will look into this ASAP
